src/Dotnet.Script/LogHelper.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Dotnet.Script.DependencyModel.Logging;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+using Microsoft.Extensions.Logging.Console.Internal;
+using LogLevel = Microsoft.Extensions.Logging.LogLevel;
+
+namespace Dotnet.Script
+{
+    public static class LogHelper
+    {                     
+        public static LogFactory CreateLogFactory(string verbosity)
+        {
+            var logLevel = (LogLevel)LevelMapper.FromString(verbosity);
+            
+            var loggerFactory = new LoggerFactory();            
+
+            loggerFactory.AddProvider(new ConsoleErrorLoggerProvider((message, level) => level >= logLevel));                        
+
+            return type =>
+            {
+                var logger = loggerFactory.CreateLogger(type);
+                return (level, message, exception) =>
+                {
+                    logger.Log((LogLevel)level, message, exception);                   
+                };
+            };
+        }    
+    }   
+    
+    public class WindowsLogErrorConsole : IConsole
+    {
+        private void SetColor(ConsoleColor? background, ConsoleColor? foreground)
+        {
+            if (background.HasValue)
+            {
+                Console.BackgroundColor = background.Value;
+            }
+
+            if (foreground.HasValue)
+            {
+                Console.ForegroundColor = foreground.Value;
+            }
+        }
+
+        private void ResetColor()
+        {
+            Console.ResetColor();
+        }
+
+        public void Write(string message, ConsoleColor? background, ConsoleColor? foreground)
+        {
+            SetColor(background, foreground);
+            Console.Error.Write(message);
+            ResetColor();
+        }
+
+        public void WriteLine(string message, ConsoleColor? background, ConsoleColor? foreground)
+        {
+            SetColor(background, foreground);
+            Console.Error.WriteLine(message);
+            ResetColor();
+        }
+
+        public void Flush()
+        {            
+            // No action required as for every write, data is sent directly to the console
+            // output stream
+        }
+    }
+
+    public class AnsiSystemErrorConsole : IAnsiSystemConsole
```

Comment:
why would we skip colors for non-windows?

Reply:
This just just a copy of an internal class.
https://github.com/aspnet/Logging/blob/c9a97e95004f27abff8012bca1ece1f2258eab90/src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs#L289

I believe the actual coloring is handled in the `AnsiLogConsole` class
https://github.com/aspnet/Logging/blob/master/src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs